include/zephyr/drivers/mfd/npm1300.h

```diff
@@ -87,6 +87,34 @@ int mfd_npm1300_reg_write2(const struct device *dev, uint8_t base, uint8_t offse
 int mfd_npm1300_reg_update(const struct device *dev, uint8_t base, uint8_t offset, uint8_t data,
 			   uint8_t mask);
 
+/**
+ * @brief Write npm1300 timer register
+ *
+ * @param dev npm1300 mfd device
+ * @param time_ms timer value in ms
+ * @retval 0 If successful
+ * @retval -EINVAL if time value is too large
+ * @retval -errno In case of any bus error (see i2c_write_dt())
+ */
+int mfd_npm1300_set_timer(const struct device *dev, uint32_t time_ms);
+
+/**
+ * @brief Enter ship mode
+ *
+ * @param dev nPM1300 MFD device instance
+ * @return Return value from i2c_write_dt
+ */
+int mfd_npm1300_ship(const struct device *dev);
```

Comment:
note: since ship mode is quite common in PMICs, shouldn't we make this a regulator-parent API (like DVS)?

Reply:
I think the decision of where to put this is a bit tricky.

It could make sense to include ship mode for all regulators, as it is very common (previous discussion here: https://github.com/zephyrproject-rtos/zephyr/pull/57998).

Hibernate is trickier as it is more specific to this device, should this be in regulator, mfd or somewhere else?

Any thoughts @danieldegrasse ?

Reply:
@gmarull My original idea was to put it in the regulator driver: https://github.com/zephyrproject-rtos/zephyr/pull/57998.
Most regulators do support ship mode, so this would seem to be reasonable.

Would you prefer I reopen and update that PR instead?